unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -138,6 +138,15 @@ public int numChars() {
   }
 
   /**
+   * Returns a 64-bit integer that can be used as the prefix used in sorting.
+   */
+  public long getPrefix() {
+    long p = PlatformDependent.UNSAFE.getLong(base, offset);
+    p = java.lang.Long.reverseBytes(p);
```

Comment:
I created https://github.com/apache/spark/pull/7789/files to make it safer. Still doesn't handle big endian though.